### DISCUSSION TITLE
Make html export work offline

### DISCUSSION THREAD

**Discussion by aszenz** [perspective-dev/perspective#2691] [Ideas]
Right now html export loads perspective libraries from jsdelivr, is it possible to inline these in the html page so that the page can work offline.

**Comment by texodus** [comment]
It is possible to inline Perspective into an HTML file without a CDN - there are already disted inline JS builds of these packages which are named with `.inline` in the filename extension. ([example for `perspective.inline.js`](https://cdn.jsdelivr.net/npm/@finos/perspective@2.10.1/dist/esm/perspective.inline.js)). However, perspective itself would still need to access a CDN somewhere to generate the downloaded page in the first place - as I'm not sure how else Perspective could get access to its own source code or the inline variant (or a separately compiled variant if the host version itself is not inline).

**Reply by aszenz** [reply]
If the host is using an inline variant, couldn't that be copied from the dom to the exported page. I guess the inline script must have some identifier for it to work.

**Reply by aszenz** [reply]
Another observation is that html export tries to load third party plugins from jsDelivr even if they are not available resulting in a 404 error.

In essence it means currently html export fails to work when custom plugins are defined for perspective viewer even if they aren't used in config.
